### PR TITLE
Semicolon a separator for search defs files. Update binary files to i…

### PR DESCRIPTION
**Pull request by michaelcfanning**
…nclude pack files. Use deny list for security rules.

@eddynaka, simple but important fixes.
@cfaucon 
@jameswinkler 
@yongyan-gh

### CODE REVIEW

**Review by eddynaka — APPROVED** [review]
:shipit:

File: Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json [hunk 1]
```diff
@@ -2,13 +2,10 @@
...
   "Definitions": [
     {
-      "SharedStrings": {
-        "$BinaryFiles": "(?i)\\.(bmp|dll|exe|gif|jpg|jpeg|lock|png|psd|tar\\.gz|tif?|ttf|xcf|zip)$",
-        "$SourceFiles": "(?i)\\.(azure|bat|c|cmd|config|cpp|cs|cscfg|definitions|dtsx|h|hxx|hpp|ini|java|jsx?|json|keys|kt|loadtest|m|md|php|properties|ps1|psm1|pubxml|py|resx|sample|sql|ste|swift|test|tsx?|txt|waz|xml)$"
-      },
       "Id": "SEC101",
       "Name": "DoNotExposePlaintextSecrets",
       "Level": "Error",
+      "FileNameDenyRegex": "$BinaryFiles",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json:8]: >FileNameDenyRegex [](start = 7, length = 17)

Oops, our security analysis was running against all files due to an incomplete earlier change.

File: Src/Plugins/Security/Security.SharedStrings.txt [hunk 2]
```diff
@@ -1 +1,24 @@
...
+# $MyExample=[a-z]
+# $MyExandedExample=($MyExample|[0-9])
+#
+# Blank lines are ignored when parsing, as are lines that begin with a '#' character.
+# Leading spaces are trimmed at parse time, allowing for indentation.
+
+# Common patterns for file allow and deny lists.
+#
+  $CSourceFiles=(?i)\.(c|cpp|cxx)$
+  $BinaryFiles=(?i)\.(bmp|dll|exe|gif|jpe?g|lock|pack|png|psd|tar\.gz|tiff?|ttf|xcf|zip)$
```
Comment (michaelcfanning) [inline Src/Plugins/Security/Security.SharedStrings.txt:21]: >pack [](start = 49, length = 4)

added pack files. also fixed the tiff? query and minimized jpg|jpeg to a single expression.

tiff? was previously rendered as tif? which matches *.ti and *.tiff. I think we want to match *.tif and *.tiff instead.

File: Src/Sarif.PatternMatcher/AnalyzeOptions.cs [hunk 3]
```diff
@@ -15,6 +15,7 @@ public class AnalyzeOptions : AnalyzeOptionsBase
         [Option(
             'd',
             "search-definitions",
+            Separator = ';',
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher/AnalyzeOptions.cs:18]: >Separator [](start = 12, length = 9)

currently to make this argument work, you need it at the very end of the command-line and you provide a space separatow. Now you can embed this arg anywhere on the conmand-line, just use the ; to separate directories. this is the separator for windows %PATH% var, so a good one, obviously for this purpose.

**Pull request merged**